### PR TITLE
ADD report HDSL method

### PR DESCRIPTION
**Pull request by robnester-rh**
Adding report() method to the HDSL. Currently it provides a single
reporting type, which is email. Recipient email addresses, cc email
addresses, bcc addresses can be supplied via contra.yml or when calling
the report method.

### CODE REVIEW

File: vars/report.groovy [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        if ( configData.bccEmail ) { combineEmailAddresses(bccEmails, configData.bccEmail) }
+        if ( config.bccEmail ){ combineEmailAddresses(bccEmails, config.bccEmail) }
+
+        // subject: JOB NAME - BUILD - RESULT
+        String subject = "${env.JOB_NAME} - ${env.BUILD_NUMBER} - ${currentBuild.currentResult}"
+        String body = """Hi, here's some info about your job!
+
+Job Name: ${env.JOB_NAME}
+Job Build #: ${env.BUILD_NUMBER}
+Jobs result: ${currentBuild.currentResult}
```
[the remark this answers is not in the document]
Comment (robnester-rh) [reply vars/report.groovy]: Updated to "Job result" in subsequent commit.

**Comment by flowerncsu** [comment]
lgtm

**Comment by arilivigni** [comment]
@robnester-rh you marked in the Jira that this is still waiting for review is that true?

**Comment by robnester-rh** [comment]
> @robnester-rh you marked in the Jira that this is still waiting for review is that true?

Nope, just dragged to the wrong column.

**Pull request merged**